design/marketplace.md

```diff
@@ -0,0 +1,279 @@
+before the timeout. The client is able to withdraw the rest of the tokens in the
+contract.
+
+A contract ends when the money that was paid upfront runs out. The end time can
+be calculated from the amount of tokens that are paid out per second. Note that
+in our scheme this amount does not change during the lifetime of the contract,
+ even when proofs are missed and repair happens. This is a desirable property
+for hosts; they can be sure of a steady source of income, and a predetermined
+contract length. When a contract ends, the hosts may withdraw their collateral.
+
+When too many hosts fail to submit storage proofs, and no other hosts take over
+the slots that they vacate, then the content can be considered lost. The
+contract is considered failed. The collateral of every host in the contract is
+burned as an additional incentive for the network hosts to avoid this scenario.
+The client is able to retrieve any funds that are left in the contract.
+
+        |
+        | create
+        |
+        v
+    -----------       timeout       -------------
+    | created | ------------------> | cancelled |
+    -----------                     -------------
+        |
+        | all slots filled
+        |
+        v
+    -----------   too many losses    ----------
+    | started | -------------------> | failed |
+    -----------                      ----------
+        |
+        | money runs out
+        |
+        v
+    ---------
+    | ended |
+    ---------
+
+
+Repairs
+-------
+
+When a slot becomes empty, the remaining collateral associated with the slot is
+used as an incentive to repair the lost content. Repair typically involves
+downloading other parts of the content and using erasure coding to restore the
+missing parts. This incurs costs for a host. To compensate the host for these
+costs it receives not only its own collateral back at the end of the contract,
+but also the remaining collateral from the host that failed to hold a slot.
+
+We expect the collateral to be significantly higher than the costs of repair.
+This means that hosts in the network can benefit greatly from repairs, and they
+may prioritize repairs over filling slots in new contracts. This is intentional,
+we want the network to prioritize honoring existing contracts over starting new
+ones.
+
+Renewal
+-------
+
+When a contract is about to end, and someone in the network wants the contract
+to continue for longer, then they can post a new contract with the same content
+identifier.
+
+We've chosen not to allow top-ups of existing contracts with new funds. Even
+though this has many advantages (it's a very simple way to extend the lifetime
+of the contract, it allows people to easily chip in to host content, etc.) it
+has one big disadvantage: hosts no longer know for how long they'll be kept to
+the contract. When a contract is continuously topped up, they cannot leave the
+contract without losing their collateral.
```

Comment:
Renewal feels like it would be something that is a very nice to have from a user experience perspective. Not allowing for renewable contracts prevents hosts from being perpetually locked in to contracts against their will. However, couldn't there be a property in the contract that a host can set allowing them to opt in or out of renewals?

Reply:
We can still offer renewal to the user; the node would simple posts a new contract with the same parameters.